### PR TITLE
Order by clause support

### PR DESCRIPTION
**Pull request by kchoy-sfdc**
Add support for ORDER BY clause using the design described in issue #3 .

### CODE REVIEW

File: marshaller.go [hunk 1]
```diff
@@ -279,6 +306,148 @@ func getReflectedValueAndType(v interface{}) (reflect.Value, reflect.Type, error
...
+		if strings.TrimSpace(fieldName) == "" {
+			return "", ErrInvalidOrderByClause
+		}
+
+		columnName, ok := columnMappings[fieldName]
+		if !ok {
+			return "", ErrInvalidOrderByClause
+		}
+
+		if tableName != "" {
```
Comment (atulkc) [inline marshaller.go:404]: Wondering when `tableName` will be non empty string? Do we support `orderBy` for nested queries (parent to child relation)?

File: marshaller.go [hunk 2]
```diff
@@ -589,6 +768,21 @@ func marshal(reflectedValue reflect.Value, reflectedType reflect.Type, childRela
 				buff.WriteString(subStr)
 			}
 		}
+		if orderByClausePresent {
+			relationName := ""
+			if childRelationName != "" {
```
Comment (atulkc) [inline marshaller.go:773]: We will need to add test for order by in child tables. If not supported, we should explicitly disallow orderBy clause for nested queries.

File: marshaller.go (hunk 1 above)
Comment (kchoy-sfdc) [reply marshaller.go:404]: SOQL supports order by clause inside nested query, that is when tableName is non-empty

File: marshaller.go (hunk 2 above)
Comment (kchoy-sfdc) [reply marshaller.go:773]: Added unit test in a followup commit

**Pull request merged**